ingestion-edge/ingestion_edge/publish.py

```diff
@@ -55,20 +56,22 @@ def _publish(topic: str, data: bytes, attrs: Dict[str, str]) -> str:
     worker.
     """
     try:
-        return client.publish(topic, data, **attrs).result()
+        return await async_wrap(client.publish(topic, data, **attrs))
```

Comment:
As a random aside, in JS `return await foo` is at best a no-op, and at worst it spins the event loop an extra time. This is because async functions mostly desugar into functions that return promises. Because of that, some style guides suggest replacing `return await foo` with `return foo` in JS (though personally I leave it to make it clearer that `foo` is an async thing.

Python doesn't work that way, and `return foo` would make a coroutine function that returns another coroutine. So `return await` is required in Python. It is interesting to compare the two languages.